WikiWatershed/model-my-watershed
# Django Upgrade

## Overview

Upgrades Django to latest long term supported version for security patches. This is the last version of Django to support Python 2.

Connects #3118 
Connects #3105 

### Notes

There are two known issues with the upgrade that were not resolved, and have been extracted into their own cards:

- Upgrade Django Rest Swagger #3124 
- Verbose Logging for 404s #3125 
- Celery `chord_unlock` Loop #3126

The Swagger update is higher priority, since currently our API docs are broken. The verbose logging is a minor annoyance. The Celery Loop is hard to recreate, but could hold up workers in production if happening consistently.

## Testing Instructions

* Check out this branch and destroy and recreate `app` and `worker`.
  - Also destroy and recreate `tiler` because the tiler service doesn't work after the first time #3112.
* Run migrations `manage.sh migrate`.
* Run `debugserver` and `debugcelery`.
  - Also start `debugtiler` to ensure it is working correctly.
* Go to [:8000/](http://localhost:8000/).
* Ensure you are not logged in.
* Draw a shape and proceed to Analyze. Ensure the analysis completes successfully.
* Go to the Monitor tab and search for items in all four catalogs. Ensure they return results correctly. Spot check the detail pages of those results to ensure they work too.
* Go to the Model tab and create a TR-55 project. Ensure it is created correctly.
* Add a new scenario, and add some changes to it.
* Open the Compare View and ensure it works correctly.
* Go back to the home page.
* Sign up for a new account. Ensure you see the email output in the console correctly. Log in with it.
* Search for a location on the map. Ensure you get suggestions. Ensure you can select a suggestion.
* Select a HUC-10 or HUC-8. Proceed to Analyze.
* Go the Model tab and create a MapShed project. Ensure it is created correctly.
* Add a new scenario. Make some changes. Ensure it works correctly.
* Go to the Water Quality tab, and select Subbasin
* Ensure subbasin results come through correctly

## Checklist
